gateway/src/node/mod.rs

```diff
@@ -130,31 +130,32 @@ where
     }
 
     pub(crate) fn print_node_details(&self) {
-        println!(
-            "Identity Key: {}",
-            self.identity_keypair.public_key().to_base58_string()
-        );
-        println!(
-            "Sphinx Key: {}",
-            self.sphinx_keypair.public_key().to_base58_string()
-        );
-        println!("Owner Signature: {}", self.generate_owner_signature());
-        println!(
-            "Host: {} (bind address: {})",
-            self.config.get_announce_address(),
-            self.config.get_listening_address()
-        );
-        println!("Version: {}", self.config.get_version());
-        println!(
-            "Mix Port: {}, Clients port: {}",
-            self.config.get_mix_port(),
-            self.config.get_clients_port()
-        );
+        let node_details = nym_types::gateway::GatewayNodeDetailsResponse {
+            identity_key: self.identity_keypair.public_key().to_base58_string(),
+            sphinx_key: self.sphinx_keypair.public_key().to_base58_string(),
+            owner_signature: self.generate_owner_signature(),
+            announce_address: self.config.get_announce_address(),
+            bind_address: self.config.get_listening_address().to_string(),
+            version: self.config.get_version().to_string(),
+            mix_port: self.config.get_mix_port(),
+            clients_port: self.config.get_clients_port(),
+            data_store: self
+                .config
+                .get_persistent_store_path()
+                .to_str()
+                .unwrap_or(".")
+                .to_string(),
+        };
 
-        println!(
-            "Data store is at: {:?}",
-            self.config.get_persistent_store_path()
-        );
+        println!("{}", node_details);
+
+        match std::fs::File::create("node_details_info.json") {
```

Comment:
shouldn't this file only be created if user passed the flag?

Reply:
also, wasn't the idea with `--output json` to print the json to `stdout` rather than writing to a file?